web/annotation_layer_builder.js

```diff
@@ -119,6 +120,61 @@ class AnnotationLayerBuilder {
       return;
     }
 
+    const uniqueLinks = inferredLinkAnnotations.filter(link => {
+      for (const annotation of annotations) {
+        const annotationRects = annot => {
```

Comment:
nit: `({ quadPoints })` and then you can remove `annot.`.

Reply:
Sorry there was an error in my code, the non-quadPoints `return` actually should have returned `[annot.rect]`, fixed it now.